force_wfmanager/wfmanager_task.py

```diff
@@ -393,30 +392,10 @@ def _bdss_done(self, exception):
             )
 
     def exit(self):
-        """ Exit the application. It first asks the user to save the current
-        Worklfow. The user can accept to save, ignore the request, or
-        cancel the quit. If the user wants to save, but the save fails, the
-        application is not closed so he has a chance to try to save again. """
-        dialog = ConfirmationDialog(
-            parent=None,
-            message='Do you want to save before exiting the Workflow '
-                    'Manager ?',
-            cancel=True,
-            yes_label='Save',
-            no_label='Don\'t save',
-        )
-
-        result = dialog.open()
-
-        if result is YES:
-            save_result = self.save_workflow()
-            if not save_result:
-                return
-        elif result is CANCEL:
-            return
+        """ Exit the main window via TaskWindowClosePrompt's window.close(),
+        which opens a dialog with the option to save the current workflow."""
 
-        app = self.window.application
-        app.exit()
+        self.window.close()
```

Comment:
I am missing the connection between the TaskWindowClosePrompt and this class. What is the magic that makes it appear when you close?